### PR TITLE
GPU hip index to smi index mapping

### PR DESCRIPTION
**Pull request by jkottiku**
Fix for erroneous power reporting in IET module

### CODE REVIEW

File: src/gpu_util.cpp [hunk 1]
```diff
@@ -34,6 +34,10 @@
 
 #include "rocm_smi/rocm_smi.h"
 #include "include/gpu_util.h"
+#define __HIP_PLATFORM_HCC__
```
Comment (frepaul) [inline src/gpu_util.cpp:37]: This is deprecated we should ne using HIP_PLATFORM_AMD

**Review by frepaul — CHANGES_REQUESTED** [review]

File: src/gpu_util.cpp (hunk 1 above)
Comment (jkottiku) [reply src/gpu_util.cpp:37]: __HIP_PLATFORM_HCC__ is present in other code as well. Need to make this change separately.

File: src/gpu_util.cpp [hunk 2]
```diff
@@ -269,19 +269,68 @@ void gpu_get_all_domain_id(std::vector<uint16_t>* pgpus_domain_id,
...
+ * @param hip_index GPU hip index
+ * @param smi_index GPU smi index
+ * @return 0 if successful, -1 otherwise
+ **/
+int gpu_hip_to_smi_index(int hip_index, uint32_t* smi_index) {
+
+  int hip_num_gpu_devices = 0;
+  uint32_t smi_num_devices = 0;
+  uint64_t val_ui64 = 0;
+  std::map<uint64_t, int> smi_map;
```
Comment (manoj-freyr) [inline src/gpu_util.cpp:300]: this can atleast be made a singleton/static to avoid repeat runs .. Approving for release block, note as todo

**Review by manoj-freyr — APPROVED** [review]
looks good,

**Pull request merged**